src/http/sr-modules-keys/sr-modules-keys.controller.ts

```diff
@@ -92,12 +92,15 @@ export class SRModulesKeysController {
 
         reply.type('application/json').send(jsonStream);
 
-        for await (const key of keysGenerator) {
-          const keyReponse = new Key(key);
-          jsonStream.write(keyReponse);
-        }
+        try {
+          for await (const key of keysGenerator) {
+            const keyReponse = new Key(key);
 
-        jsonStream.end();
+            jsonStream.write(keyReponse);
+          }
+        } finally {
+          jsonStream.end();
```

Comment:
It's good to add re-throwing error here because the code here is executed inside transaction and if error happens (and if it's silent) the transaction will not rollback.

Reply:
Nevermind, here we are reading the data, not writing it.